### PR TITLE
T28736 Re-enable parental controls tests

### PR DESCRIPTION
**Pull request by pwithnall**
A bunch of fixes to re-enable and update the parental controls tests.

This includes a couple of other fixes for general tests. I can split those out if review of the parental controls tests drags on.

The structure of the tests isn’t changed much. I changed `parental_controls_setup` to log in as the new child user and disable the desktop background, notifications, etc., so that it’s possible to log in as them and reliably match needles (just like we do for the main user normally).

Happy to answer questions about the justifications for any of the changes here: it’s taken about a week (on and off) to put this together, so it’s probably not particularly coherent.

 * Passing `parental_controls_setup` test: https://openqa.endlessm.com/tests/57954
 * Passing `parental_controls_gnome_software_oars` test: https://openqa.endlessm.com/tests/58140

https://phabricator.endlessm.com/T28736

### CODE REVIEW

**Comment by mwleeds** [comment]
This needs to be rebased on master. I tried to do that but hit merge conflicts. As is GitHub sees this PR as trying to merge commits that are already on master, such as "_graphical_login: Type the password more slowly".

**Comment by pwithnall** [comment]
Oops, rebased.

File: lib/installedtest.pm [hunk 1]
```diff
@@ -39,12 +39,14 @@ sub user_console {
     # Switch to a default or specified TTY and log in as a non-root user.
     # Use exit_user_console() to exit.
     my $self = shift;
```
Comment (mwleeds) [inline lib/installedtest.pm]: Don't know what `$self` is for, but maybe that's just my lack of Perl knowledge

File: needles/control_center_users_locked.json [hunk 2]
```diff
@@ -12,9 +12,15 @@
     },
     {
       "": "Unlock button",
+<<<<<<< HEAD
```
Comment (mwleeds) [inline needles/control_center_users_locked.json]: merge conflicts to be resolved

File: lib/installedtest.pm (hunk 1 above)
Comment (mwleeds) [reply lib/installedtest.pm]: Never mind, it's just the `installedtest` object

File: tests/parental_controls_gnome_software_oars.pm [hunk 3]
```diff
@@ -85,17 +84,14 @@ sub run {
     # Try installing Cow’s Revenge from the command line. This should also fail.
     check_desktop_clean();
     type_very_safely("gnome-terminal\n");
-    assert_screen('desktop_terminal', 5);
+    sleep(5);
     type_string("gnome-software --install system/flatpak/flathub/desktop/com.github.dariasteam.cowsrevenge/stable\n");
```
Comment (mwleeds) [inline tests/parental_controls_gnome_software_oars.pm]: Would it make more sense to use `install_app` so we're testing that the Flatpak CLI enforces parental controls?

File: needles/control_center_users_locked.json (hunk 2 above)
Comment (pwithnall) [reply needles/control_center_users_locked.json]: What? git didn’t tell me about this! argh.

File: tests/parental_controls_gnome_software_oars.pm (hunk 3 above)
Comment (pwithnall) [reply tests/parental_controls_gnome_software_oars.pm]: No, this test is specifically to check that gnome-software doesn’t show an inappropriate app when run from the command line.

File: needles/control_center_users_locked.json (hunk 2 above)
Comment (pwithnall) [reply needles/control_center_users_locked.json]: It seems I have somehow completely messed up the rebase and need to re-test this all. Bear with me.

Comment (pwithnall) [reply needles/control_center_users_locked.json]: OK, ready for review again. Sorry about that.

**Comment by pwithnall** [comment]
Fixed the rebase. Latest passing tests:
 * https://openqa.endlessm.com/tests/58584
 * https://openqa.endlessm.com/tests/58588

**Pull request merged**